docs/blog/2017-07-19-creating-a-blog-with-gatsby/index.md

```diff
@@ -0,0 +1,486 @@
+}) {
+  const { markdownRemark: post } = data; // data.markdownRemark holds our post data
+  return (
+    <div className="blog-post-container">
+      <Helmet title={`Your Blog Name - ${post.frontmatter.title}`} />
+      <div className="blog-post">
+        <h1>{post.frontmatter.title}</h1>
+        <div className="blog-post-content" dangerouslySetInnerHTML={{ __html: post.html }} />
+      </div>
+    </div>
+  );
+}
+```
+
+Whoa, neat! This React component will be rendered to a static HTML string (for each route/blog post we define), which will serve as the basis of our routing/navigation for our blog.
+
+At this point, there is a reasonable level of confusion and "magic" occuring, particularly with the props injection. What is `markdownRemark`? Where is this `data` prop injected from? All good questions, so let's answer them by writing a GraphQL query to seed our `<Template />` component with content!
+
+### Writing the GraphQL query
+
+Below the `Template` declaration, we'll want to add a GraphQL query. This is an incredibly powerful utility provided by Gatsby which lets us pick and choose very simply the pieces of data that we want to display for our blog post. Each piece of data our query selects will be injected via the `data` property we specified earlier.
+
+```javascript{21-32}
+import React from 'react';
+import Helmet from 'react-helmet';
+
+// import '../css/blog-post.css';
+
+export default function Template({
+  data
+}) {
+  const { markdownRemark: post } = data;
+  return (
+    <div className="blog-post-container">
+      <Helmet title={`Your Blog Name - ${post.frontmatter.title}`} />
+      <div className="blog-post">
+        <h1>{post.frontmatter.title}</h1>
+        <div className="blog-post-content" dangerouslySetInnerHTML={{ __html: post.html }} />
+      </div>
+    </div>
+  );
+}
+
+export const pageQuery = graphql`
+  query BlogPostByPath($path: String!) {
+    markdownRemark(frontmatter: { path: { eq: $path } }) {
+      html
+      frontmatter {
+        date(formatString: "MMMM DD, YYYY")
+        path
+        title
+      }
+    }
+  }
+`;
+```
+
+If you're not familar with GraphQL, this may seem slightly confusing, but we can break down what's going down here piece by piece.
+
+*Note: To learn more about GraphQL, consider this [excellent resource][learn-graphql]*
+
+The underlying query name `BlogPostByPath` (note: these query names need to be unique!) will be injected with the current path, e.g. the specific blog post we are viewing. This path will be available as `$path` in our query. For instance, if we were viewing our previously created blog post, the path of the file that data will be pulled from will be `/hello-world`.
+
+`markdownRemark` is the main data pull, and each item we pull from the underlying data store will be available to our React component as `data.markdownRemark.someValue`, e.g. `data.markdownRemark.html` will be the injected HTML content transformed from the original Markdown source.
```

Comment:
"main data pulled into the component". The sentence is a bit awkward though and could be rewritten further.